tests/queries/0_stateless/01915_merge_prewhere_virtual_column_rand_chao_wang.sql

```diff
@@ -9,7 +9,10 @@ ENGINE = MergeTree()
 ORDER BY f1;
 
 -- In version 20.12 this query sometimes produces an exception "Cannot find column"
-SELECT f2 FROM merge(currentDatabase(), '^abc$') PREWHERE _table = 'abc' AND f1 = 'a' AND rand() % 100 < 20;
-SELECT f2 FROM merge(currentDatabase(), '^abc$') PREWHERE _table = 'abc' AND f1 = 'a';
+SELECT f2 FROM merge(currentDatabase(), '^abc$') PREWHERE _table = 'abc' AND f1 = 'a' AND rand() % 100 < 20; -- { serverError ILLEGAL_PREWHERE }
```

Comment:
This test is strange, since there is no data blocks in the table (added in #25390 for #23163 issue)

It does not work before either:

```
CREATE TABLE abc (     `f1` String,     `f2` String ) ENGINE = MergeTree ORDER BY f1;
insert into abc values ('abc', 'abc');
SELECT * FROM merge(currentDatabase(), '^abc$') PREWHERE _table = 'abc';
Code: 10. DB::Exception: Not found column _table in block f1 String String(size = 0), f2 String String(size = 0). (NOT_FOUND_COLUMN_IN_BLOCK)
```

I'm OK to put some backward incompatible label, though AFAICS after this patch, only more cases with PREWHERE over Merge is allowed.